src/koopmans/utils/_xml.py

```diff
@@ -60,4 +60,7 @@ def read_xml_array(xml_file: Path, norm_const: float, string: str = 'EFFECTIVE-P
                 array_xml[k, j, i] = rho_tmp[(j % (nr_xml[1]-1))*(nr_xml[0]-1)+(i % (nr_xml[0]-1))]
     array_xml *= norm_const
 
-    return array_xml[:-1, :-1, :-1]
+    if in_xsf_format: # the xsf format requires an array where the last element is equal to the first element in each dimension
+        return array_xml
+    else:
+        return array_xml[:-1, :-1, :-1]
```

Comment:
Correct me if I'm wrong, but if the xsf format requires an array where the last element is the same as the first element, then we should be discarding the last element if `in_xsf_format` is `True`, not `False`, no?

Reply:
For the `array_xml` it holds that the last element is equal to the first element, e.g. 
`array_xml[-1,j,i]=array_xml[0,j,i]`.
Hence, `array_xml` is already in the correct format for xsf. But for all other calculation we don't want the extra boundary terms that don't contain additional information.

Reply:
Ah I see, I think I was confused by the purpose of the `in_xsf_format` argument. I was interpreting it as "read an xml file in xsf format" rather than "read an xml file and return an object suitable to be used in an xsf file". Can you please add a longer docstring to the function explaining each argument (something that really I ought to be doing throughout)? Optionally, we could also change the argument name to something like `retain_final_element` to avoid this sort of confusion.

Reply:
Yes, that's a good idea, thanks.

Reply:
Thanks! Can you please just make sure we don't have any lines > 120 characters long and then we're good to commit